commons-packet/commons-packet-manager/src/main/java/io/mosip/commons/packet/facade/PacketReader.java

```diff
@@ -1,190 +1,264 @@
-        return getProvider(source, process).validatePacket(id, source, process);
-    }
-
-    private IPacketReader getProvider(String source, String process) {
-        LOGGER.info(PacketManagerLogger.SESSIONID, PacketManagerLogger.REGISTRATIONID, null,
-                "getProvider for source : " + source + " process : " + process);
-        IPacketReader provider = null;
-        if (referenceReaderProviders != null && !referenceReaderProviders.isEmpty()) {
-            Optional<IPacketReader> refProvider = referenceReaderProviders.stream().filter(refPr ->
-                    (PacketHelper.isSourceAndProcessPresent(refPr.getClass().getName(), source, process, PacketHelper.Provider.READER))).findAny();
-            if (refProvider.isPresent() && refProvider.get() != null)
-                provider = refProvider.get();
-        }
-
-        if (provider == null) {
-            LOGGER.error(PacketManagerLogger.SESSIONID, PacketManagerLogger.REGISTRATIONID, null,
-                    "No available provider found for source : " + source + " process : " + process);
-            throw new NoAvailableProviderException();
-        }
-
-        return provider;
-    }
-
-}
+package io.mosip.commons.packet.facade;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+import io.mosip.commons.khazana.spi.ObjectStoreAdapter;
+import io.mosip.commons.packet.constants.PacketUtilityErrorCodes;
+import io.mosip.commons.packet.dto.Document;
+import io.mosip.commons.packet.dto.TagResponseDto;
+import io.mosip.commons.packet.exception.GetTagException;
+import io.mosip.commons.packet.exception.NoAvailableProviderException;
+import io.mosip.commons.packet.exception.ObjectStoreAdapterException;
+import io.mosip.commons.packet.spi.IPacketReader;
+import io.mosip.commons.packet.util.PacketHelper;
+import io.mosip.commons.packet.util.PacketManagerLogger;
+import io.mosip.kernel.biometrics.entities.BiometricRecord;
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.kernel.core.exception.BaseUncheckedException;
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+
+/**
+ * The packet Reader facade
+ */
+@RefreshScope
+@Component
+public class PacketReader {
+
+    private static final Logger LOGGER = PacketManagerLogger.getLogger(PacketReader.class);
+
+	@Value("${packet.manager.account.name}")

```

Comment:
private String PACKET_MANAGER_ACCOUNT;

	@Autowired
	@Qualifier("SwiftAdapter")
	private ObjectStoreAdapter swiftAdapter;

	@Autowired
	@Qualifier("S3Adapter")
	private ObjectStoreAdapter s3Adapter;

	@Autowired
	@Qualifier("PosixAdapter")
	private ObjectStoreAdapter posixAdapter;

	@Value("${objectstore.adapter.name}")
	private String adapterName;



**Same logic is present in keeper also. Please move this logic to a single place**